core/src/main/scala/org/apache/spark/sql/execution/columnar/encoding/ColumnEncoding.scala

```diff
@@ -634,7 +634,7 @@ trait ColumnEncoder extends ColumnEncoding {
     }
     val baseTypeOffset = offset(position).toInt
     // initialize the null bytes to zeros
-    allocator.clearBuffer(columnData, baseTypeOffset, numNullBytes)
+    allocator.fill(columnData, 0, baseTypeOffset, numNullBytes)
```

Comment:
Memsetting to '0' for the full buffer. Did it cause a problem or what? clearBuffer does the same thing too. Right?

Reply:
fill is a more generic method so store code was refactored a bit to remove clearBuffer and instead use fill. The recent addition being of "spark.memory.debugFill" property now honoured by store's BufferAllocator too to fill initialized/freed memory with non-zero bytes (so that any code errors of reading from those will lead to incorrect results)